app/src/screens/PersonCredentialLoading.tsx

```diff
@@ -0,0 +1,258 @@
+      textAlign: 'center',
+      marginTop: 30,
+    },
+    controlsContainer: {
+      marginTop: 'auto',
+      margin: 20,
+    },
+    delayMessageText: {
+      textAlign: 'center',
+      marginTop: 20,
+    },
+  })
+
+  useEffect(() => {
+    const connect = async () => {
+      try {
+        const remoteAgentDetails = await connectToIASAgent(agent, store, t)
+        setRemoteAgentDetails(remoteAgentDetails)
+        logger.error(`Connected to IAS agent, connectionId: ${remoteAgentDetails.connectionId}`)
+      } catch (err) {
+        logger.error(`Failed to connect to IAS agent, error: ${(err as BifoldError).message}`)
+      }
+    }
+    connect()
+  }, [])
+
+  useEffect(() => {
+    if (!remoteAgentDetails) {
+      return
+    }
+
+    timer.current = setTimeout(() => {
+      const proofRequest = receivedProofRequests.find((proof) => proof.connectionId === remoteAgentDetails.connectionId)
+
+      if (!proofRequest) {
+        // No proof from our IAS Agent to respond to, do nothing.
+        logger.info(`Waited ${attestationProofRequestWaitTimeout / 1000}sec on attestation proof request, continuing`)
+
+        setDidCompleteAttestationProofRequest(true)
+      }
+    }, attestationProofRequestWaitTimeout)
+  }, [remoteAgentDetails])
+
+  // Use this function to accept the attestation proof request.
+  const acceptAttestationProofRequest = async (agent: BifoldAgent, proofRequest: ProofExchangeRecord) => {
+    logger.info('Attestation: selecting credentials for attestation proof request')
+    // This will throw if we don't have the necessary credentials
+    const credentials = await agent.proofs.selectCredentialsForRequest({
+      proofRecordId: proofRequest.id,
+    })
+
+    logger.info('Attestation: accepting attestation proof request')
+    await agent.proofs.acceptRequest({
+      proofRecordId: proofRequest.id,
+      proofFormats: credentials.proofFormats,
+    })
+
+    return true
+  }
+
+  // when a person credential offer is received, show the
+  // offer screen to the user.
+  const goToCredentialOffer = (credentialId?: string) => {
+    navigation.getParent()?.navigate(Stacks.NotificationStack, {
+      screen: Screens.CredentialOffer,
+      params: { credentialId },
+    })
+  }
+
+  useEffect(() => {
+    const handleFailedAttestation = (error: BifoldError) => {
+      navigation.goBack()
+      DeviceEventEmitter.emit(BifoldEventTypes.ERROR_ADDED, error)
+    }
+
+    const subscriptions = Array<EmitterSubscription>()
+    subscriptions.push(DeviceEventEmitter.addListener(AttestationEventTypes.FailedHandleProof, handleFailedAttestation))
+    subscriptions.push(DeviceEventEmitter.addListener(AttestationEventTypes.FailedHandleOffer, handleFailedAttestation))
+    subscriptions.push(
+      DeviceEventEmitter.addListener(AttestationEventTypes.FailedRequestCredential, handleFailedAttestation)
+    )
+
+    return () => {
+      subscriptions.forEach((subscription) => subscription.remove())
+    }
+  }, [])
```

Comment:
I think in this one you may need:

- The **navigation** object is included in the dependencies array to ensure that the effect re-runs if navigation changes.
- **DeviceEventEmitter, BifoldEventTypes, or AttestationEventTypes** are constants and do not change, they don't need to be in the dependencies array. If they can change, they should be added to the array as well.

Reply:
Not quite sure what you mean with adding the navigation object to the dependancy array. The navigation object should only change when we trigger a navigation event like navigating to a new screen, in which case we wouldn't want to re-run the useEffect because we'd be leaving this screen right?